csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -953,12 +953,8 @@ private module Cached {
       callCfn = any(Call c | isParamsArg(c, _, _)).getAControlFlowNode()
     } or
     TFlowInsensitiveFieldNode(FieldOrProperty f) { f.isFieldLike() } or
-    TInstanceParameterAccessNode(ControlFlow::Node cfn, boolean isPostUpdate) {
-      exists(ParameterAccess pa | cfn = getAPrimaryConstructorParameterCfn(pa) |
-        isPostUpdate = false
-        or
-        pa instanceof ParameterWrite and isPostUpdate = true
-      )
+    TInstanceParameterAccessNode(ControlFlow::Node cfn, Boolean isPostUpdate) {
+      cfn = getAPrimaryConstructorParameterCfn(_)
```

Comment:
Why was this changed? It seems that we are now creating redundant post-update nodes for parameter reads.

Reply:
Without post-update nodes this lead to a failing dataflow consistency test for (according to the `reverseRead`)
```csharp
class ClassC2(object oc2) : ClassC1(oc2) { }
```
Should we add these as exceptions?